File: src/main/java/com/back/pinco/global/exception/ValidationField.java
package com.back.pinco.global.exception;

public sealed interface ValidationField
        permits LatitudeField, LongitudeField, ContentField, UnknownField {

    String name();

    static ValidationField from(String fieldName) {
        return switch (fieldName) {
            case "latitude" -> new LatitudeField();
            case "longitude" -> new LongitudeField();
            case "content" -> new ContentField();
            default -> new UnknownField(fieldName);
        };
    }
}

final class LatitudeField implements ValidationField {
    @Override public String name() { return "latitude"; }
}

final class LongitudeField implements ValidationField {
    @Override public String name() { return "longitude"; }
}

final class ContentField implements ValidationField {
    @Override public String name() { return "content"; }
}

final class UnknownField implements ValidationField {
    private final String name;
    public UnknownField(String name) { this.name = name; }
    @Override public String name() { return name; }
}

File: src/main/java/com/back/pinco/global/exception/ValidationField.kt
package com.back.pinco.global.exception

sealed interface ValidationField {

    companion object {
        fun from(fieldName: String): ValidationField = when (fieldName) {
                "latitude" -> LatitudeField
                "longitude" -> LongitudeField
                "content" -> ContentField
                else -> UnknownField(fieldName)
        }
    }

    fun name(): String
}

data object LatitudeField : ValidationField {
    override fun name(): String = "latitude"
}

data object LongitudeField : ValidationField {
    override fun name(): String = "longitude"
}

data object ContentField : ValidationField {
    override fun name(): String = "content"
}

data class UnknownField(private val name: String) : ValidationField {
    override fun name() : String = name
}
